### PR TITLE
upgrade i18next to latest version

### PR DESCRIPTION
**Pull request by zoran995**
### What this PR does

Fixes #<insert issue number here if relevant>

Upgrade i18next to the latest version, we still can't use new pluralization format as weblate still doesn't support it

### Checklist

-   ~[ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~
-   ~[ ] I've updated relevant documentation in `doc/`.~
-   [x] I've updated CHANGES.md with what I changed.
-   ~[ ] I've provided instructions in the PR description on how to test this PR.~

### CODE REVIEW

File: yarn.lock
Comment (staffordsmith83) [inline yarn.lock:967]: Starting to get quite a few duplicates here. Can we run `yarn-deduplicate` , or upgrade to yarn v2.x, I think it has builtin de-duplication? Here is output of `yarn why`:

```
=> Found "@babel/runtime@7.17.7"
info Has been hoisted to "@babel/runtime"
info Reasons this module exists
   - "workspace-aggregator-6660f4f1-831c-4eef-adee-5a4964da2e5a" depends on it
   - Hoisted from "_project_#terriajs#rc-slider#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-i18next#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-select#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-transition-group#@babel#runtime"
   - Hoisted from "_project_#terriajs#rc-slider#rc-tooltip#@babel#runtime"
   - Hoisted from "_project_#terriajs#rc-slider#rc-util#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-select#@emotion#core#@babel#runtime"
   - Hoisted from "_project_#@babel#preset-env#@babel#plugin-transform-regenerator#regenerator-transform#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-transition-group#dom-helpers#@babel#runtime"
   - Hoisted from "_project_#terriajs#rc-slider#rc-tooltip#rc-trigger#@babel#runtime"
   - Hoisted from "_project_#terriajs#rc-slider#rc-tooltip#rc-trigger#rc-align#@babel#runtime"
   - Hoisted from "_project_#terriajs#rc-slider#rc-tooltip#rc-trigger#rc-motion#@babel#runtime"
   - Hoisted from "_project_#terriajs#react-select#@emotion#css#babel-plugin-emotion#babel-plugin-macros#@babel#runtime"
info Disk size without dependencies: "844KB"
info Disk size with unique dependencies: "888KB"
info Disk size with transitive dependencies: "888KB"
info Number of shared dependencies: 1
=> Found "fetch-mock#@babel/runtime@7.17.9"
info This module exists because "_project_#terriajs#fetch-mock" depends on it.
info Disk size without dependencies: "844KB"
info Disk size with unique dependencies: "888KB"
info Disk size with transitive dependencies: "888KB"
info Number of shared dependencies: 1
=> Found "i18next-browser-languagedetector#@babel/runtime@7.18.9"
info This module exists because "_project_#terriajs#i18next-browser-languagedetector" depends on it.
info Disk size without dependencies: "880KB"
info Disk size with unique dependencies: "924KB"
info Disk size with transitive dependencies: "924KB"
info Number of shared dependencies: 1
=> Found "i18next#@babel/runtime@7.18.9"
info This module exists because "_project_#terriajs#i18next" depends on it.
info Disk size without dependencies: "880KB"
info Disk size with unique dependencies: "924KB"
info Disk size with transitive dependencies: "924KB"
info Number of shared dependencies: 1
✨  Done in 0.84s.
```

File: yarn.lock
Comment (staffordsmith83) [inline yarn.lock:11748]: Is it definitely OK to upgrade React from 16.3.2 to 16.14.0 ?

File: CHANGES.md
Comment (staffordsmith83) [inline CHANGES.md]: Doesnt the i18next.init() in this PR specify to still use v3? https://github.com/zoran995/terriajs-i18n/blob/cf5d7a1ff1043e1fbbd1e26dcf4392c3cdb2cd96/lib/Models/Internationalization.ts#L86

File: lib/Models/Internationalization.ts
Comment (staffordsmith83) [inline lib/Models/Internationalization.ts]: Is this OK to set to false?
"If your interpolation variables are user provided or loaded from an external source, we strongly suggest to keep this option to true.
If you know what you're doing, you can also set this to false."
from https://www.i18next.com/translation-function/interpolation

File: lib/Models/Internationalization.ts
Comment (staffordsmith83) [inline lib/Models/Internationalization.ts:157]: I assume `checkWhitelist: true` removed becasue this is the default anyway?

File: lib/Models/Internationalization.ts
Comment (staffordsmith83) [inline lib/Models/Internationalization.ts:158]: I assume `checkWhitelist: true` removed becasue this is the default anyway?

File: yarn.lock
Comment (staffordsmith83) [reply yarn.lock:11748]: All good

File: lib/Models/Internationalization.ts
Comment (staffordsmith83) [reply lib/Models/Internationalization.ts]: I see that this is the default behaviour pre v21, and we were previously using v19, so this is just retaining the same settings.

File: yarn.lock
Comment (staffordsmith83) [reply yarn.lock:967]: Would like to hear your opinion, but I can put this is a separate ticket for something that could be done repo-wide, wont hold up approval.

**Review by staffordsmith83 — CHANGES_REQUESTED** [review]
Hi @zoran995 thanks again for your hard work.
A few small comments and then will approve when I hear back from you, cheers!

File: CHANGES.md
Comment (zoran995) [reply CHANGES.md]: Good catch, forgot to revert this in changes.md, we have to wait with upgrade of files to v4 until weblate supports them

File: lib/Models/Internationalization.ts
Comment (zoran995) [reply lib/Models/Internationalization.ts:158]: Yeah commit message on i18next detection don't even mention this change they just say respect values set in main i18next config. And as I figured out they are now doing this by default

Comment (zoran995) [reply lib/Models/Internationalization.ts:158]: also in i18next-browser-languageDetector v3.1.1 `checkWhitelist` was set to true by default

File: lib/Models/Internationalization.ts
Comment (zoran995) [reply lib/Models/Internationalization.ts]: Actually, we are safe to use skipOnVariables as true, I just went through i18next tests (only those cases that are documented won't be interpolated) and don't see it anywhere in the codebase. Thanks for raising this once again

Comment (zoran995) [reply lib/Models/Internationalization.ts]: changed it to use a default value

File: yarn.lock
Comment (zoran995) [reply yarn.lock:967]: We probably should have it as a separate ticket as it should be done both TerriaJS and TerriaMap, yarn is on v3 now :D

Comment (staffordsmith83) [reply yarn.lock:967]: OK Ill make one :-)

**Review by staffordsmith83 — APPROVED** [review]
All looks resolved to me, thanks @zoran995

**Pull request merged**
